src/AggregateFunctions/DDSketch/Store.h

```diff
@@ -227,8 +227,14 @@ class DenseStore : public Store
         }
         else
         {
-            int new_length = getNewLength(new_min_key, new_max_key);
-            bins.resize(new_length, 0.0);
+            UInt32 new_length = getNewLength(new_min_key, new_max_key);
+            if (new_length > bins.capacity())
+            {
+                bins.reserve(new_length);
+            }
+            std::size_t old_size = bins.size();
+            bins.resize(new_length);
+            std::fill(bins.begin() + old_size, bins.end(), 0.0);
```

Comment:
The `if (new_length > bins.capacity())` doesn't change anything, `reserve()` doesn't do anything if requested length is less than current capacity.
The `std::fill()` doesn't do anything because `resize()` already initializes the new elements with default values.

So this simplifies to:
```
            UInt32 new_length = getNewLength(new_min_key, new_max_key);
            bins.reserve(new_length);
            bins.resize(new_length);
```
I.e. adding a reserve() before resize() to disable exponential growth.

Just in case you didn't understand my previous proposal, it was the opposite:
```
bins.resize(new_length);

// (At this point bins.capacity() is usually greater than bins.size(), because resize() grows capacity() exponentially.
//  such that a series of increasing resize()s takes O(final_size) time, even if resizing by one element at a time.
//  But here we do an additional O(n) pass in adjust(), so exponentially growing capacity() doesn't help,
//  we're still O(n^2).
//  So we make size() grow exponentially together with capacity(), by resizing again.)

bins.resize(bins.capacity());
```
(Or maybe I missed some obvious reason why this doesn't work?)

If you understood and decided against it - that's fine.

Reply:
No, I misunderstood.

> if (new_length > bins.capacity()) doesn't change anything

I thought your suggestion was to target the case where new length is more than current capacity.

> because resize() already initializes the new elements with default values.

I didn't know about this. I thought the new elements wouldn't contain default values unless explicitly mentioned. Let me make the change and test once.